### DISCUSSION TITLE
[ZK Stack] - WebSocket for local dockerized node not working

### DISCUSSION THREAD

**Discussion by Szczepoo13** [zkSync-Community-Hub/zksync-developers#769] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [X] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.83.0-nightly

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

Hey all,

I'm trying to flood a network with transactions by executing a specified number of WETH-DAI swaps via a WebSocket RPC endpoint.
I use a local dockerized node from the zkstack running my chain.
While performing these swaps, I'm encountering a 403 Forbidden error. I'm using the WebSocket URL 'ws://127.0.0.1:3050'.

My question is: could the WebSocket URL be incorrect? I couldn't find any documentation on the WebSocket URL for a dockerized node. Also, what could be causing the 403 access error?

Thank you! 😊

### Expected Behavior

I am able to access web sockets and perform swaps through it

### Current Behavior

While performing these swaps, I'm encountering a 403 Forbidden error. I'm using the WebSocket URL 'ws://127.0.0.1:3050'.

### Prior Research

- [ ] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It seems like you are facing an issue with accessing the WebSocket RPC endpoint on your local dockerized ZKsync node. The WebSocket URL you are using, `ws://127.0.0.1:3050`, appears to be correct as per the standard configuration for local setups. However, the 403 Forbidden error typically indicates a permissions issue.

Here are a few steps you can check to resolve this issue:

1. **Ensure WebSocket Service is Enabled**: Verify that the WebSocket service is enabled in your dockerized node's configuration. This might involve checking the docker-compose file or any relevant configuration files where services are defined.

2. **Check Network Configuration**: Ensure that there are no network configurations or firewall settings that might be blocking WebSocket connections.

3. **Authentication and Permissions**: Check if there are any authentication or permissions required to access the WebSocket endpoint that you might not have set up.

4. **Logs and Debugging**: Review the logs for your dockerized node to see if there are any additional details provided about why the access is forbidden.

If you continue to face issues, you might want to revisit the setup instructions or check for any updates that might affect WebSocket connections in the ZKsync dockerized environment. For more detailed discussions and support, you can visit the [ZKsync developers discussions on Github](https://github.com/zkSync-Community-Hub/zksync-developers/discussions).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by popzxc** [comment, marked as answer]
3050 is the port for the HTTP API. For WebSocket API you need port 3051.
